src/platform/nrfconnect/FactoryDataParser.h

```diff
@@ -58,11 +58,17 @@ struct FactoryData
     uint32_t passcode;
     struct FactoryDataString enable_key;
     struct FactoryDataString user;
+    struct ProductAppearance
+    {
+        uint8_t finish;
+        uint8_t primary_color;
```

Comment:
This should really be `Optional<uint8_t>` for the color, if you want to support "finish and maybe color".

Reply:
Ok, right, I forgot that this is not required.